pkg/build/api/validation/validation.go

```diff
@@ -171,6 +178,10 @@ func validateSource(input *buildapi.BuildSource, isCustomStrategy, isDockerStrat
 		}
 	}
 
+	if input.Git == nil && isJenkinsPipelineStrategyFromRepo {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("git"), "", "must be set when using Jenkins Pipeline strategy with Jenkinsfile from a git repo"))
```

Comment:
@bparees OK, then I was confused because I thought that field can be used to override the Jenkinsfile filename :-)